WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -601,8 +581,18 @@ - (void)setFollowing:(BOOL)following forPostsFromSiteWithID:(NSNumber *)siteID a
  */
 - (WordPressComRestApi *)apiForRequest
 {
-    WPAccount *defaultAccount = [WPAccount lookupDefaultWordPressComAccountInContext:self.managedObjectContext];
+    WordPressComRestApi * __block api = nil;
+    [self.coreDataStack.mainContext performBlockAndWait:^{
+        api = [self apiForRequestInContext:self.coreDataStack.mainContext];
+    }];
+    return api;
+}
+
+- (WordPressComRestApi *)apiForRequestInContext:(NSManagedObjectContext *)context
+{
+    WPAccount *defaultAccount = [WPAccount lookupDefaultWordPressComAccountInContext:context];
     WordPressComRestApi *api = [defaultAccount wordPressComRestApi];
+
     if (![api hasCredentials]) {
         api = [WordPressComRestApi defaultApiWithOAuthToken:nil
                                                   userAgent:[WPUserAgent wordPressUserAgent]
```

Comment:
Out of scope, but keen to hear what you think.

This logic is more or less duplicated between `ReaderPostService.m`, `ReaderSiteService.m`, and `ReaderTopicService.m`. I wonder if we could extract it 🤔 

<img width="409" alt="image" src="https://user-images.githubusercontent.com/1218433/220812420-44872e7c-c8a6-4fdf-a959-7b5c0b801c94.png">

Reply:
This logic of creating an instance of `WordPressComRestApi` is duplicated in lots of places, not just these types used in the Reader. They are all implemented in the similar way, but probably have about 4 or 5 subtly different implementation. It's probably worth extracting them at some point.

Reply:
Tracked in https://github.com/wordpress-mobile/WordPress-iOS/issues/20208